tests/e2e/specFiles/settings.spec.ts

```diff
@@ -0,0 +1,111 @@
+import { expect } from 'detox';
+
+import { isBeta, WAIT_FOR_ELEMENT_TIMEOUT } from '../helpers';
+import app from '../pageObjects';
+import { SupportedLanguage } from '../pageObjects/pages/settings/LanguageScreen';
+
+describe('Settings', () => {
+  beforeEach(async () => {
+    isBeta() && (await app.onboarding.betaVersionScreen.close());
+    await app.developerRoom.tapOnSkipOnboardingButton();
+    await app.navigationBar.changeTab('settings');
+  });
+
+  describe('@android @ios @smoke', () => {
+    it('should display Settings screen correctly', async () => {
+      expect(app.settings.settingsScreen.goldWalletIcon).toBeVisible();
+    });
+  });
+
+  describe('@android @ios @regression', () => {
+    it('should be possible to turn on advanced options', async () => {
+      await app.settings.settingsScreen.tapOnAdvancedOptions();
+      await app.settings.advancedOptionsScreen.tapOnAdvancedOptionsSwitch();
+
+      await app.header.tapOnBackButton();
+      await app.navigationBar.changeTab('wallets');
+
+      await app.wallets.dashboardScreen.tapOnAddButton();
+
+      await expect(app.wallets.addNewWallet.createScreen.walletTypeRadios['Standard HD P2SH']).toBeVisible();
+      await expect(app.wallets.addNewWallet.createScreen.walletTypeRadios['Standard HD SegWit']).toBeVisible();
+      await expect(app.wallets.addNewWallet.createScreen.walletTypeRadios['Standard P2SH']).toBeVisible();
+    });
+
+    it('should be possible change PIN', async () => {
+      await app.settings.settingsScreen.tapOnChangePin();
+
+      await app.settings.changePin.currentPinScreen.typePin('1234');
+      await app.settings.changePin.newPinScreen.typePin('1111');
+      await app.settings.changePin.confirmPinScreen.typePin('1111');
+
+      // TODO: Add more assertions
+      await expect(app.settings.changePin.successScreen.icon).toBeVisible();
+    });
+
+    it('should display an error validation message if typed current PIN is incorrect', async () => {
+      await app.settings.settingsScreen.tapOnChangePin();
+      await app.settings.changePin.currentPinScreen.typePin('0000');
+
+      await expect(app.settings.changePin.currentPinScreen.pinValidationError).toBeVisible();
+    });
+
+    it('should show "Application blocked" screen if number of attemps is exceeded', async () => {
+      await app.settings.settingsScreen.tapOnChangePin();
+
+      await app.settings.changePin.currentPinScreen.typePin('0000');
```

Comment:
[?] What do you think about hiding this string under one variable, I think it will be more manageable?

Reply:
Not really, at least not in this case. It's a string that I want to use explicitly here. Not sure if it'll be beneficial to have, for example, `invalidPin` variable instead of raw `0000`.

Reply:
Okay, up to you.